contracts/BC_fusion/StakeCredit.sol

```diff
@@ -0,0 +1,249 @@
+        return _totalBnbAmount;
+    }
+
+    function distributeReward(uint64 commissionRate) external payable onlyStakeHub {
+        uint256 bnbAmount = msg.value;
+        uint256 _commission = (bnbAmount * uint256(commissionRate)) / COMMISSION_RATE_BASE;
+        uint256 _reward = bnbAmount - _commission;
+        totalPooledBNB += _reward;
+
+        // mint commission to the validator
+        _mintAndSync(validator, _commission);
+
+        emit RewardReceived(_reward, _commission);
+    }
+
+    function slash(uint256 slashBnbAmount) external onlyStakeHub returns (uint256) {
+        uint256 selfDelegation = balanceOf(validator);
+        uint256 slashShares = getSharesByPooledBNB(slashBnbAmount);
+
+        slashShares = slashShares > selfDelegation ? selfDelegation : slashShares;
+        uint256 realSlashBnbAmount = _burnAndSync(validator, slashShares);
+        emit Slashed(realSlashBnbAmount);
+
+        uint256 _gasLimit = IStakeHub(STAKE_HUB_ADDR).transferGasLimit();
+        (bool success,) = SYSTEM_REWARD_ADDR.call{ gas: _gasLimit, value: realSlashBnbAmount }("");
+        require(success, "TRANSFER_FAILED");
+        return realSlashBnbAmount;
+    }
+
+    /*----------------- view functions -----------------*/
+    /**
+     * @return the amount of shares that corresponds to `_bnbAmount` protocol-controlled BNB.
+     */
+    function getSharesByPooledBNB(uint256 bnbAmount) public view returns (uint256) {
+        if (totalPooledBNB == 0) {
+            return 0;
+        }
+        return (bnbAmount * totalSupply()) / totalPooledBNB;
+    }
+
+    /**
+     * @return the amount of BNB that corresponds to `_sharesAmount` token shares.
+     */
+    function getPooledBNBByShares(uint256 shares) public view returns (uint256) {
+        if (totalSupply() == 0) {
+            return 0;
+        }
+        return (shares * totalPooledBNB) / totalSupply();
+    }
+
+    function unbondRequest(address delegator, uint256 _index) public view returns (UnbondRequest memory, uint256) {
+        bytes32 hash = _unbondRequestsQueue[delegator].at(_index);
+        return (_unbondRequests[hash], _unbondRequestsQueue[delegator].length());
+    }
+
+    function lockedShares(address delegator) public view returns (uint256) {
+        return _lockedShares[delegator];
+    }
+
+    function unbondSequence(address delegator) public view returns (uint256) {
+        return _unbondSequence[delegator].current();
+    }
+
+    function getSelfDelegationBNB() public view returns (uint256) {
+        return getPooledBNBByShares(balanceOf(validator));
+    }
+
+    function getPooledBNB(address account) external view returns (uint256) {
+        return getPooledBNBByShares(balanceOf(account));
+    }
+
+    /*----------------- internal functions -----------------*/
+    function _bootstrapInitialHolder(uint256 initAmount) internal onlyInitializing {
+        assert(validator != address(0));
```

Comment:
sometimes you use "require", sometimes "assert", can we unify it?

Reply:
Done